### PR TITLE
Eahw2601/timecard changes

### PR DESCRIPTION
**Pull request by Xeena-Newman**

### CODE REVIEW

File: src/components/timecard/edit-shift-timecard/EditShiftTimecard.jsx [hunk 1]
```diff
@@ -68,9 +69,18 @@ const EditShiftTimecard = ({
...
           </dt>
-          <dd className="govuk-summary-list__value"></dd>
+          <dd
+            className="govuk-summary-list__value"
+            style={{ whiteSpace: 'nowrap' }}
+          >
+            {timeEntryExists
+              ? timecardDate === formatDate(timeEntry.startTime)
+                ? 'Shift'
+                : 'Shift (continued)'
```
Comment (daleelR) [inline src/components/timecard/edit-shift-timecard/EditShiftTimecard.jsx:81]: MIght be worth using the getting the 'Shift' text from the map returned by the `getTimePeriodTypesMap` function as that comes from the API. Easier maintenance.

**Review by IanFisherHO — CHANGES_REQUESTED** [review]

File: src/components/timecard/edit-shift-timecard/EditShiftTimecard.spec.jsx
Comment (daleelR) [inline src/components/timecard/edit-shift-timecard/EditShiftTimecard.spec.jsx:1]: Can we have tests to cover the new logic that shows "HH:MM **on DD Month** to" when the shift covers multiple days.
And also tests to cover the logic of conditionally showing the 'continued' text on shifts that cover multiple days.

File: src/components/timecard/timecard-entries-list/TimecardEntriesList.spec.jsx [hunk 2]
```diff
@@ -22,7 +22,7 @@ describe('TimecardEntriesList', () => {
     expect(screen.queryByText('Add a new time period')).toBeFalsy();
     expect(screen.getByText('Start time')).toBeTruthy();
     expect(screen.getByText('Finish time')).toBeTruthy();
-    expect(screen.getByText('Shift')).toBeTruthy();
```
Comment (daleelR) [inline src/components/timecard/timecard-entries-list/TimecardEntriesList.spec.jsx:25]: Could you not still keep this since the test is specifically testing for when the period type is shift?

File: src/pages/timecard/Timecard.spec.jsx [hunk 3]
```diff
@@ -80,6 +80,48 @@ describe('Shift spanning mutiple days', () => {
...
+        },
+      ],
+      setTimeEntries: jest.fn(),
+      timecardDate: '',
+      setTimecardDate: jest.fn(),
+    });
+
+    expect(
+      screen.getByText((content) =>
+        content.includes('01:00 on 1 September to 05:00 on 3 September')
```
Comment (daleelR) [inline src/pages/timecard/Timecard.spec.jsx:101]: I think this can conflict with the test name. The test name suggests that it just needs to display the start and finish time when they have both been entered.
But this particular text we'd only want to show when covering multiple days. 
Maybe the test name should be updated.

File: src/pages/timecard/Timecard.spec.jsx [hunk 4]
```diff
@@ -80,6 +80,48 @@ describe('Shift spanning mutiple days', () => {
...
 
+  it('should display start and finish time on timecard when both times have been entered', async () => {
+    renderWithTimecardContext(<Timecard />, {
+      summaryErrors: {},
+      setSummaryErrors: jest.fn(),
+      timeEntries: [
+        {
+          timePeriodTypeId: '00000000-0000-0000-0000-000000000001',
+          startTime: '2022-09-01 01:00:00+00:00',
+          finishTime: '2022-09-03 05:00:00+00:00',
```
Comment (daleelR) [inline src/pages/timecard/Timecard.spec.jsx:91]: Assuming this test is actually specifically to cover those covering multiple days, it might be better to test the boundaries of the logic/requirement by having the start time be 11:59 and the finish time 00:00 the following day.

File: src/pages/timecard/Timecard.spec.jsx [hunk 5]
```diff
@@ -80,6 +80,48 @@ describe('Shift spanning mutiple days', () => {
...
+    });
+
+    expect(
+      screen.getByText((content) =>
+        content.includes('01:00 on 1 September to 05:00 on 3 September')
+      )
+    ).toBeTruthy();
+  });
+
+  it('should display Shift(continued) if timecard spans more than one date and we are not on start date', async () => {
```
Comment (daleelR) [inline src/pages/timecard/Timecard.spec.jsx:106]: Can we add a similar specific test to show that continued doesn't show when it spans more that one day but we're on the start date.

File: src/pages/timecard/Timecard.spec.jsx (hunk 3 above)
Comment (Xeena-Newman) [reply src/pages/timecard/Timecard.spec.jsx:101]: my mistake, that's been changed now

File: src/pages/timecard/Timecard.spec.jsx (hunk 4 above)
Comment (Xeena-Newman) [reply src/pages/timecard/Timecard.spec.jsx]: that's been changed now

File: src/pages/timecard/Timecard.spec.jsx (hunk 5 above)
Comment (Xeena-Newman) [reply src/pages/timecard/Timecard.spec.jsx:106]: That's been added now

File: src/components/timecard/timecard-entries-list/TimecardEntriesList.spec.jsx (hunk 2 above)
Comment (daleelR) [reply src/components/timecard/timecard-entries-list/TimecardEntriesList.spec.jsx:25]: Ah ok, yes that makes sense, was just curious. Will resolve.

File: src/components/timecard/edit-shift-timecard/EditShiftTimecard.jsx (hunk 1 above)
Comment (Xeena-Newman) [reply src/components/timecard/edit-shift-timecard/EditShiftTimecard.jsx]: a change has been made...it's caused 17 tests to fail, but is this change you meant?

Comment (Xeena-Newman) [reply src/components/timecard/edit-shift-timecard/EditShiftTimecard.jsx]: This will be revisited as a sub-task after the ui revamp

Comment (daleelR) [reply src/components/timecard/edit-shift-timecard/EditShiftTimecard.jsx]: Yeah that was the change, resolving after discussion and should be picked up as part of the revamp.

**Review by IanFisherHO — APPROVED** [review]

File: src/pages/timecard/Timecard.spec.jsx (hunk 5 above)
Comment (daleelR) [reply src/pages/timecard/Timecard.spec.jsx:106]: Might be a bit better to also expect that "(continued)" is falsy in the new test.

**Review by daleelR — APPROVED** [review]
